### ISSUE TITLE
Windows MSVC wrong or misspelled target value will fail to compile

### ISSUE THREAD

**Issue by mrcdk** [godotengine/godot#11562]
**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Godot 3.0 master https://github.com/godotengine/godot/commit/1f4685375f2d2e36a2e19e67ac9f614b8c99ea28

**Issue description:**

If you misspell or write the `target` value wrong the compilation will fail.

`target=debug` Compiles
`target=release_debug` Compiles
`target=debug_release` Fails to compile with the error below
`target=this_is_broken_pls_fix` Fails to compile with the error below

**Steps to reproduce:**
Compile with a wrong `target` on windows using msvc

[Compilation log](https://hastebin.com/joxotijoci.erl)

**Comment by akien-mga** [comment]
Relevant part:
```
modules\gdnative\nativescript\api_generator.cpp(340): error C2039: 'get_enum_list': is not a member of 'ClassDB'
z:\cpp\godot\core\class_db.h(90): note: see declaration of 'ClassDB'
modules\gdnative\nativescript\api_generator.cpp(340): error C3861: 'get_enum_list': identifier not found
modules\gdnative\nativescript\api_generator.cpp(345): error C2039: 'get_enum_constants': is not a member of 'ClassDB'
z:\cpp\godot\core\class_db.h(90): note: see declaration of 'ClassDB'
modules\gdnative\nativescript\api_generator.cpp(345): error C3861: 'get_enum_constants': identifier not found
```

**Comment by akien-mga** [comment]
cc @karroffel

**Comment by ISylvox** [comment]
Try to compile with release_debug instead of debug_release ? I usually build Godot with target=release_debug (no issue on compiling https://github.com/godotengine/godot/commit/09800ac65079599c568679b53962a313182885ea )

**Comment by mrcdk** [comment]
Right, I made I mistake. It's `release_debug` not `debug_release` 😅 
I've compiled https://github.com/godotengine/godot/commit/09800ac65079599c568679b53962a313182885ea correctly with `target=release_debug` so that's still working and wasn't broken to begin with My question now is why `target=debug_release` is broken? Shouldn't it default to `target=debug`? Why then `target=debug` isn't broken? 😕 

I've just tested from a clean state from this commit https://github.com/godotengine/godot/commit/09800ac65079599c568679b53962a313182885ea :
`target=debug` Compiles
`target=release_debug` Compiles
`target=debug_release` Fails to compile with the error before

**Comment by akien-mga** [comment]
There is no `target=debug_release`. What's your exact command?

**Comment by mrcdk** [comment]
I've updated the issue. It happens when the `target` value is wrong. `scons platform=windows target=this_is_broken -j8` will also throw the same error.

**Comment by akien-mga** [comment]
Wait there is actually a `target=debug_release` on Windows, I had never noticed that.

**Comment by akien-mga** [comment]
The issue with no target being specified should be fixed by #11567.

**Comment by mrcdk** [comment]
Closing this issues because https://github.com/godotengine/godot/pull/11567 fixed it.

Thanks guys.
